external/js/cothority/src/network/connection.ts

```diff
@@ -53,7 +54,12 @@ export class WebSocketConnection implements IConnection {
      * @param service   Name of the service to reach
      */
     constructor(addr: string, service: string) {
-        this.url = addr;
+        const url = new URL(addr, {});
+        if (typeof window !== "undefined" && window.isSecureContext === true) {
```

Comment:
I'm surprised you don't get a reference error here, as it's not defined in a NodeJS environment. Have you run the tests locally ?

Reply:
> I'm surprised you don't get a reference error here, as it's not defined in a NodeJS environment.

Notice the `typeof window` guard to ensure its existence.

> Have you run the tests locally ?

Yup, I did.

Reply:
oh right